client/js/src/client.ts

```diff
@@ -1042,7 +1041,14 @@ export function api_factory(
 							)
 						);
 					} else if (event_callbacks[event_id]) {
-						await event_callbacks[event_id](_data);
+						if (_data.msg === "process_completed") {
+							unclosed_events.delete(event_id);
+							if (unclosed_events.size === 0) {
+								close_stream();
+							}
+						}
+						let fn = event_callbacks[event_id];
+						window.setTimeout(fn, 0, _data);
```

Comment:
Because SSE onmessage events take high precedence by browsers (higher than websocket onmessage events), they block the browser's rendering if many events are received in quick succession. Moving the event callback to happen after a timeout places the callback in the JS event queue, which reduces its priority to happen after a browser render. 
However, closing the event stream does need to happen immediately if a process_complete is received, otherwise the browser will think the connection broke when the backend closes the stream, so we moved that logic here so it happens in the onmessage immediately and not later in the timeout.

Reply:
Makes sense. Ish.